modules/social_features/social_group/social_group.module

```diff
@@ -2205,44 +2205,33 @@ function social_group_user_cancel_methods_alter(&$methods) {
 /**
  * Implements hook_social_user_account_header_create_links().
  *
- * Adds the "Create Group" link to the content creation menu.
- * When the user can only create groups of one type, the user skips the
- * group type selection page.
+ * Adds the "Create Group" link to the content creation menu, that redirect user
+ * to the create form of flexible group. If the user doesn't have access the
+ * link won't be available.
  */
 function social_group_social_user_account_header_create_links($context) {
-
-  // If we have a user to work with, we can check if we should redirect it to
-  // the create group overview or a single group.
-  // If we do not have a user we'll default to the overview.
-  $route_add_group = NULL;
   if (!empty($context['user'])) {
     /** @var \Drupal\Core\Session\AccountInterface $user */
     $account = $context['user'];
     /** @var \Drupal\Core\Url|NULL $route_add_group */
     $route_add_group = \Drupal::service('social_group.helper_service')
       ->getGroupsToAddUrl($account);
 
-    // Get groups to add url can return Url service
-    // It will change it when necessary.
+    // Get group to add url can return Url service.
     if ($route_add_group instanceof Url) {
       $route_add_group = Link::fromTextAndUrl((new TranslatableMarkup('New Group')), $route_add_group);
+      return [
+        'add_group' => [
+          '#type' => 'link',
+          '#attributes' => [
+            'title' => new TranslatableMarkup('Create New Group'),
+          ],
+          '#weight' => 500,
+          '#access' => $route_add_group->getUrl()->access(),
+        ] + $route_add_group->toRenderable(),
+      ];
```

Comment:
This indentation makes it look like a lot of code has changed, but we really only moved code from being after [a guard clause](https://betterprogramming.pub/refactoring-guard-clauses-2ceeaa1a9da) to being inside a similar if-statement.

Preserving the guard clause and ensuring we don't need to indent unchanged code makes reviewing a lot easier :) 

There's the added problem that this new version has a situation in which the function return `void` even though `social_user.api.php` clearly requires the function always returns an array.